plugins/woocommerce-admin/client/products/sections/product-shipping-section.tsx

```diff
@@ -52,53 +72,212 @@ export const ProductShippingSection: React.FC = () => {
+						'woocommerce_weight_unit',
+					] ),
+			};
+		},
+		[]
+	);
+
+	const inputWidthProps = getTextControlProps(
+		getInputProps( 'dimensions.width' )
+	);
+	const inputLengthProps = getTextControlProps(
+		getInputProps( 'dimensions.length' )
+	);
+	const inputHeightProps = getTextControlProps(
+		getInputProps( 'dimensions.height' )
+	);
+	const inputWeightProps = getTextControlProps( getInputProps( 'weight' ) );
+
 	return (
-		<ProductSectionLayout
-			title={ __( 'Shipping', 'woocommerce' ) }
-			description={ __(
-				'Set up shipping costs and enter dimensions used for accurate rate calculations.',
-				'woocommerce'
-			) }
-		>
-			{ hasResolvedShippingClasses ? (
-				<div>
-					<SelectControl
-						label={ __( 'Shipping class', 'woocommerce' ) }
-						{ ...getTextControlProps(
-							getInputProps( 'shipping_class' )
+		<>
+			<ProductSectionLayout
+				title={ __( 'Shipping', 'woocommerce' ) }
+				description={ __(
+					'Set up shipping costs and enter dimensions used for accurate rate calculations.',
+					'woocommerce'
+				) }
+			>
+				<Card>
+					<CardBody className="product-shipping-section__classes">
+						{ hasResolvedShippingClasses ? (
+							<>
+								<SelectControl
+									label={ __(
+										'Shipping class',
+										'woocommerce'
+									) }
+									{ ...getTextControlProps(
+										getInputProps( 'shipping_class' )
+									) }
+									options={ [
+										...DEFAULT_SHIPPING_CLASS_OPTIONS,
+										...mapShippingClassToSelectOption(
+											shippingClasses ?? []
+										),
+									] }
+								/>
+								<span className="woocommerce-product-form__secondary-text">
+									{ interpolateComponents( {
+										mixedString: __(
+											'Manage shipping classes and rates in {{link}}global settings{{/link}}.',
+											'woocommerce'
+										),
+										components: {
+											link: (
+												<Link
+													href={ `${ ADMIN_URL }admin.php?page=wc-settings&tab=shipping&section=classes` }
+													target="_blank"
+													type="external"
+												>
+													<></>
+												</Link>
+											),
+										},
+									} ) }
+								</span>
+							</>
+						) : (
+							<div className="product-shipping-section__spinner-wrapper">
+								<Spinner />
+							</div>
+						) }
+					</CardBody>
+				</Card>
+
+				<Card>
+					<CardBody className="product-shipping-section__dimensions">
+						{ hasResolvedUnits ? (
+							<>
+								<h4 className="product-shipping-section__subtitle">
+									{ __( 'Dimensions', 'woocommerce' ) }
+								</h4>
+								<p className="woocommerce-product-form__secondary-text">
+									{ __(
+										'Enter the size of the product as you’d put it in a shipping box, including packaging like bubble wrap.',
+										'woocommerce'
+									) }
+								</p>
+								<div className="product-shipping-section__container">
+									<div>
+										<BaseControl
```

Comment:
I don't think the `BaseControl` is needed for wrapping each `InputControl`. The logic within `BaseControl` is already integrated within the `InputControl` if you look at the label code for example:
BaseControl: https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/base-control/index.tsx#L52-L73
InputControl: https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/input-control/label.tsx

Was this recommended somewhere else?

Reply:
> Other TextControls we are using they are all wrapped within the BaseControl as you can see [here](https://github.com/WordPress/gutenberg/blob/trunk/packages/components/src/text-control/index.tsx#L40).

Yeah, although the `InputControl` is suppose to replace this `TextControl` so maybe the `help` feature needed as part of your second point will still get added to the `InputControl`. I guess we should keep an eye out during any WordPress updates for this.

I think it would be worth to make our own `InputControl` wrapper that acts much like the `TextControl` you referenced, that wraps the `BaseControl` already. So in the case were WordPress does add this on update we only have one spot to update, and not a bunch of places.
But I acknowledge this would be out of the scope of this PR and can be done as a follow up.